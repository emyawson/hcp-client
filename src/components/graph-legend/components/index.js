export * from './legend-item.component';
export * from './detail-graph-legend.component';
export * from './logbook-graph-legend.component';
export * from './logbook-stats-graph-legend.component';
export * from './trend-graph-legend.component';
export * from './metabolic-graph-legend.component';
export * from './insulin-pump-legend.component';
