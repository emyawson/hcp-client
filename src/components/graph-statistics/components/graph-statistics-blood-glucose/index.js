export * from './graph-statistics-blood-glucose.component';
