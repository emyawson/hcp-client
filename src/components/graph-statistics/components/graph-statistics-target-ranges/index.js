export * from './graph-statistics-target-ranges.component';
