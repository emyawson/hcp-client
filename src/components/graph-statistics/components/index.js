export * from './graph-statistics-blood-glucose';
export * from './graph-statistics-target-ranges';
export * from './metabolic-statistics-tests.component';
export * from './metabolic-statistics-blood-glucose.component';
export * from './metabolic-statistics-indexes.component';
