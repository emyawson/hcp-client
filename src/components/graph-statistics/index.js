export * from './graph-statistics.component';
