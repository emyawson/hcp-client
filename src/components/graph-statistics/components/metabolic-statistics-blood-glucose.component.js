import React from 'react';

import { LocalizedText } from 'src/components';

import {
  ListSection,
  ListsContainer,
  List,
  ListElement,
  ListElementSectionHeader,
} from '../graph-statistics.style';

export const MetabolicStatistcsBloodGlucose = ({
  mean,
  meanBeforeMeal,
  meanAfterMeal,
  stdDev,
  stdDevMeanRatio,
}) => (
  <ListSection borderRight>
    <ListElementSectionHeader>
      <LocalizedText textKey="graphDetails.statistics.bloodGlucose.bloodGlucose" />
    </ListElementSectionHeader>
    <ListsContainer>
      <List flex={'none'}>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.bloodGlucose.meanBloodGlucose" />
        </ListElement>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.bloodGlucose.meanBloodGlucoseBeforeMeal" />
        </ListElement>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.bloodGlucose.meanBloodGlucoseAfterMeal" />
        </ListElement>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.bloodGlucose.standardDeviationSD" />
        </ListElement>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.bloodGlucose.stdDevMeanRatio" />
        </ListElement>
      </List>
      <List>
        <ListElement>
          {isNaN(mean) || mean === 0 ? ' - ' : mean + ' '}
          <LocalizedText textKey="graphDetails.mgPerDL" />
        </ListElement>
        <ListElement>
          {isNaN(meanBeforeMeal) || meanBeforeMeal === 0
            ? ' - '
            : meanBeforeMeal + ' '}
          <LocalizedText textKey="graphDetails.mgPerDL" />
        </ListElement>
        <ListElement>
          {isNaN(meanAfterMeal) || meanAfterMeal === 0
            ? ' - '
            : meanAfterMeal + ' '}
          <LocalizedText textKey="graphDetails.mgPerDL" />
        </ListElement>
        <ListElement>
          {`${stdDev} `}
          <LocalizedText textKey="graphDetails.mgPerDL" />
        </ListElement>
        <ListElement>
          {isNaN(stdDevMeanRatio) ? ' - ' : stdDevMeanRatio + ' '}%
        </ListElement>
      </List>
      <List flex={8} />
    </ListsContainer>
  </ListSection>
);
