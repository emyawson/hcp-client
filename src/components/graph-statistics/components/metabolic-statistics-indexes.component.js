import React from 'react';

import { LocalizedText } from 'src/components';

import {
  ListSection,
  ListsContainer,
  List,
  ListElement,
  ListElementSectionHeader,
} from '../graph-statistics.style';

export const MetabolicStatistcsIndexes = ({ lbgi, hbgi }) => (
  <ListSection>
    <ListElementSectionHeader>
      <LocalizedText textKey="graphDetails.statistics.indexes.indexes" />
    </ListElementSectionHeader>
    <ListsContainer>
      <List flex={'none'}>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.indexes.hbgi" />
        </ListElement>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.indexes.lbgi" />
        </ListElement>
      </List>
      <List>
        <ListElement>{hbgi}</ListElement>
        <ListElement>{lbgi}</ListElement>
      </List>
      <List flex={8} />
    </ListsContainer>
  </ListSection>
);
