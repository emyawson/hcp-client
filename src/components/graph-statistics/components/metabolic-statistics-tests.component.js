import React from 'react';

import { LocalizedText } from 'src/components';

import {
  ListSection,
  ListsContainer,
  List,
  ListElement,
  ListElementSectionHeader,
} from '../graph-statistics.style';

export const MetabolicStatistcsTests = ({
  numberOfTests,
  testsPerDay,
  testsPerMeasuredDay,
}) => (
  <ListSection borderRight>
    <ListElementSectionHeader>
      <LocalizedText textKey="graphDetails.statistics.tests.tests" />
    </ListElementSectionHeader>
    <ListsContainer>
      <List flex={'none'}>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.tests.numberOfTests" />
        </ListElement>
        <ListElement bold>
          <LocalizedText textKey="graphDetails.statistics.tests.testsPerDay" />
        </ListElement>
      </List>
      <List>
        <ListElement>{`${numberOfTests || '0'} `}</ListElement>
        <ListElement>
          {testsPerDay === '0.0' ? ' - ' : testsPerDay}
          ({isNaN(testsPerMeasuredDay) ? ' - ' : testsPerMeasuredDay})
        </ListElement>
      </List>
      <List flex={8} />
    </ListsContainer>
  </ListSection>
);
