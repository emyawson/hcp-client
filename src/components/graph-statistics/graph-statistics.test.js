import React from 'react';
import { shallow } from 'enzyme';

import { GraphDetail } from 'src/components';

const mockProps = {
  graphDetails: {
    bloodGlucoseValues: {
      bloodGlucoseMean: 158.4,
      bloodGlucoseStandardDeviation: 137.6,
      testsPerDay: 10,
    },
    targetRangesValues: {
      abovePercentage: 40,
      belowPercentage: 20,
      hypoglycaemiaNumber: 3,
      hypoglycaemiaPercentage: 30,
      targetBloodGlucoseMinimum: 80,
      targetBloodGlucoseMaximum: 125,
      withinPercentage: 20,
    },
  },
  graphStartTime: '0:00',
};

describe('graph-statistics test suite', () => {
  test('it renders correctly', () => {
    const wrapper = shallow(
      <GraphDetail {...mockProps}>
        <div>test</div>
      </GraphDetail>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
