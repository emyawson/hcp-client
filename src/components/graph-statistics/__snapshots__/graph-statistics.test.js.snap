// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graph-statistics test suite it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    graphDetails={
      Object {
        "bloodGlucoseValues": Object {
          "bloodGlucoseMean": 158.4,
          "bloodGlucoseStandardDeviation": 137.6,
          "testsPerDay": 10,
        },
        "targetRangesValues": Object {
          "abovePercentage": 40,
          "belowPercentage": 20,
          "hypoglycaemiaNumber": 3,
          "hypoglycaemiaPercentage": 30,
          "targetBloodGlucoseMaximum": 125,
          "targetBloodGlucoseMinimum": 80,
          "withinPercentage": 20,
        },
      }
    }
    graphStartTime="0:00"
  >
    <div>
      test
    </div>
  </Unknown>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "minHeight": "12rem",
      "navItemLabelSelectedStyle": "
  path,
  circle {
    fill: #0D79D6;
  }
  .horizontalLines {
    stroke: #0D79D6;
  }
  color: #0D79D6;
",
      "tabs": Array [
        Object {
          "content": <Unknown
            graphDetails={undefined}
            graphType={undefined}
          />,
          "keyText": "statistics",
          "label": "%",
        },
        Object {
          "content": <Unknown
            graphType={undefined}
          />,
          "keyText": "legend",
          "label": <Unknown />,
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "minHeight": "12rem",
        "navItemLabelSelectedStyle": "
  path,
  circle {
    fill: #0D79D6;
  }
  .horizontalLines {
    stroke: #0D79D6;
  }
  color: #0D79D6;
",
        "tabs": Array [
          Object {
            "content": <Unknown
              graphDetails={undefined}
              graphType={undefined}
            />,
            "keyText": "statistics",
            "label": "%",
          },
          Object {
            "content": <Unknown
              graphType={undefined}
            />,
            "keyText": "legend",
            "label": <Unknown />,
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
